llvm/include/llvm/SandboxIR/SandboxIR.h

```diff
@@ -553,6 +557,50 @@ class Instruction : public sandboxir::User {
 #endif
 };
 
+class LoadInst : public Instruction {
+  /// Use Context::createLoadInst(). Don't call the
+  /// constructor directly.
+  LoadInst(llvm::LoadInst *LI, Context &Ctx)
+      : Instruction(ClassID::Load, Opcode::Load, LI, Ctx) {}
+  friend Context; // for LoadInst()
+  Use getOperandUseInternal(unsigned OpIdx, bool Verify) const final {
+    return getOperandUseDefault(OpIdx, Verify);
+  }
+  SmallVector<llvm::Instruction *, 1> getLLVMInstrs() const final {
+    return {cast<llvm::Instruction>(Val)};
+  }
+
+public:
+  unsigned getUseOperandNo(const Use &Use) const final {
```

Comment:
The `final` confused me. It comes from `User` . For style and safeness, I would mark it as `override` first.

Reply:
Hmm why would you mark it as `override` ? `LoadInst` won't have subclasses so it should be safe to mark it `final`.

Reply:
True. But `getUseOperandNo` overrides a virtual = 0 from `User`. Once you mark it `override` the compiler will do checks for you. `override final`.

Reply:
I think `override final` is redundant, and I am not sure it's being used in LLVM. If you use `final` the compiler will check that you are actually overriding and print an error if not, and on top of that it will also check that it is final.

Reply:
Indeed. I found occurrence.

Reply:
Now that `LoadInst` is final, the finals are kind of redundant. Feel free to ignore, but `override` would be better.

Reply:
Aren't they still useful though? If you misspell a `final` function name you will still get an error, but you won't get one if you just rely on the class being marked `final`.
Well, I think I would prefer to stick with `final` because it conveys both that it's overriden and that it's final.

Reply:
No worries. The class is marked final. You cannot inherit from anyway and overwriting the functions. My only confusion is that final is kind of redundant. `override` is less confusing and gives the same compiler checks.